src/java.desktop/unix/classes/sun/awt/X11/XEmbeddingContainer.java

```diff
@@ -130,19 +132,23 @@ public void dispatchEvent(XEvent xev) {
 
     void forwardKeyEvent(long child, KeyEvent e) {
         byte[] bdata = AWTAccessor.getAWTEventAccessor().getBData(e);
-        long data = Native.toData(bdata);
-        if (data == 0) {
+        if (bdata == null) {
             return;
         }
-        XKeyEvent ke = new XKeyEvent(data);
-        ke.set_window(child);
-        XToolkit.awtLock();
-        try {
-            XlibWrapper.XSendEvent(XToolkit.getDisplay(), child, false, XConstants.NoEventMask, data);
-        }
-        finally {
-            XToolkit.awtUnlock();
+        try (Arena arena = Arena.ofConfined()) {
+            var data = arena.allocateFrom(ValueLayout.JAVA_BYTE, bdata);
+            try {
+                XKeyEvent ke = new XKeyEvent(data.address());
+                ke.set_window(child);
+                XToolkit.awtLock();
+                try {
+                    XlibWrapper.XSendEvent(XToolkit.getDisplay(), child, false, XConstants.NoEventMask, data.address());
+                } finally {
+                    XToolkit.awtUnlock();
+                }
+            } finally {
+                Reference.reachabilityFence(data);
```

Comment:
Can you explain the need for this `reachabilityFence()`? I don't understand how `data` could become unreachable before the `Arena` is closed. Also, it's an in-scope local variable. Thanks!

Reply:
You are right. As we have the arena,  we do not need to invoke the reachabilityFence.

Reply:
Yes, I had asked the same question !